### ISSUE TITLE
Choose method to use

### ISSUE THREAD

**Issue by e-caste** [WEEE-Open/cannolo#1]
The kickstart file is a flexible way to do some scripting and automate installation. Would need to allow choice (or add later) between BIOS and UEFI installation.

The preseed.cfg is similar to the kickstart file. It doesn't seem to work at the moment, but with a little work it may be the most efficient way to do this.

The golden image expansion is to look into (and the original method we thought of).

**Comment by morpheusthewhite** [comment]
The best (and working) technique to solve the problem seem to be the golden image.

- I did very few tests on the kickstart file because of its limited support in Ubuntu: I succeeded only to select the right option in the installation menu. However each step was still presented on the screen
- The `pressed.cfg` is the recommended technique to perform an unattended installation on ubuntu. It also seems to have a good and complete documentation. It seems. Because according to [some opinions](https://askubuntu.com/questions/1002043/how-to-an-unattended-installation-of-ubuntu-16-04-on-a-disk-with-existing-os) it is limited to UEFI mode while according to [others](http://gyk.lt/ubuntu-16-04-desktop-unattended-installation/) it seems that presseding can only be used on Ubuntu Server images. [This project](https://github.com/core-process/linux-unattended-installation) automates the process of automation: by customizing the preseeding file an Ubuntu Desktop installation can be chosen, but it needs an Internet connection and it increases the installation time. 
- The golden image offers the possibility to create a customized `.img` file to replicate: [PiShrink](https://github.com/Drewsif/PiShrink) is a similar project developed for Raspberry which offers a similar approach for the image shrinking process. Extension of the primary partition can be then easily solved with `growroot` and `resize2fs`. I tested this solution on a MBR partition and it works, but it still needs to be tested on GPT partition schemes and LVM (which can probably offer more flexibility).
